include/swift/Basic/SourceManager.h

```diff
@@ -166,6 +167,10 @@ class SourceManager {
     return FileSystem;
   }
 
+  void setEditorMode() {
+    editorMode = true;
+  }
+
   void setIDEInspectionTarget(unsigned BufferID, unsigned Offset) {
```

Comment:
Does this function being called imply that we should set the editor mode as well?

Reply:
I'm not sure. I think probably this function will only be used within running within the context of an IDE, but IMO it is cleaner to only set editor mode in one place, during the setup of the` CompilerInstance`.